patterns/Behavioral/Observal/JavaScript/README.md

```diff
@@ -1,3 +1,122 @@
-# ✨ JavaScript - Observal
+# ✨ JavaScript Function Implementation of Observer Pattern
+
+## Introduction
+
+The Observer pattern is a behavioral design pattern that lets objects subscribe to events and receive notifications when those events happen. The independent 
+
+object (known as a subject) notifies other objects (observers) when its state changes. This pattern is commonly used in JavaScript to implement event handling systems
+
+It's fundamental to event-driven programming, especially in JavaScript, promoting flexible and loosely coupled design.
+
+Take a look at the diagram below.
+
+## Diagram
+
+![Observer Pattern](/patterns/Behavioral/Observal/JavaScript/images/javascript-observer.jpg)
+
+The Subject has three methods to subscribe, unsubscribe, and notify its list of Observers.
+The Observers are interested in knowing about the Subject's state changes, like button click events. So they must subscribe to the Subject to be notified. When a user clicks the button, the observers update the user interface to reflect the user activity.
+
+Consider the Facebook Page as an example. Facebook Page acts as the Subject which has a list of followers (observers) who wish to receive updates. And it has the following methods:
+- Subscribe: A user subscribes by Following
+- Unsubscribe: They can unsubscribe by Unfollowing the page
+- Notify: It notifies all followers (observers) whenever there is a new post and every time a user engages with it. Then the user's UI updates in real-time to show the post and comments on their feed.
+
+Now, let's dive deeper into this pattern with the below discussion on its key aspects.
+
+1. Subject Implementation
+Subject (Observable):
+- The `ButtonSubject` constructor represents the Subject.
```

Comment:
Lines 27 and 28 are not rendering the way you intend, I don't think. Both appear on the same line in the rendered markdown. You need a line break between them in the raw markdown (which won't be rendered) if you want them to be on separate lines.